Allow plugin's input_type without prefix ext__

vtran99 opened the pull request:
Request this change to allow plugins to use custom input_type name without the prefix 'ext__' although the registration form custom field still be named 'ext__<name>'.
(without this change, the InputComponent for the custom field can't be found in FormItems)

ThiefMaster commented:
Hadn't we agreed to allow registering a component without the `ext__` prefix as long as something like `force: true` is specified? I remember such a discussion on Matrix not too long ago...

ThiefMaster commented:
Yes, check the thread @foxbunny started on April 4 where I suggested doing something like this:

```js
  if (
    Object.items(pluginFields).some(
      ([name, data]) =>
        !name.startsWith('ext__') || (fieldRegistry[name] && data.unsafeOverrideField)
    )
  ) {
    throw new Error(
      'Field names from plugins must begin with `ext__` or match an existing field name and set ' +
        'the `unsafeOverrideField` property to true'
    );
```

vtran99 commented:
I updated the code following your suggestion:
I removed the check "fieldRegistry[name]" to allow the new component even if there is no existing field with same name: case adding a new input type ...

ThiefMaster commented:
> I removed the check "fieldRegistry[name]" to allow the new component even if there is no existing field with same name: case adding a new input type ...

The check was there for a reason: People should NOT add new field names without the `ext__` prefix, because we do not want cases where us adding a new input type results in a conflict with someone's custom component.

By allowing the override without the prefix ONLY for components that already exist in the core, this is avoided.

Is there any reason why you do not want to use the prefix for your custom fields, and only use the extra attribute to allow overriding a core field when needed?

vtran99 commented:
> Is there any reason why you do not want to use the prefix for your custom fields, and only use the extra attribute to allow overriding a core field when needed?

No, it's just for convenience and make code less "weird" when reading:
I'm adding a component Birth Date with a new input type "birthdate", with current constraint I have to name the input type "ext__birthdate",
then in our code when we want to check if a field is birth date input, we will need to check:
`input_type == 'ext__birthdate'`
with seems a bit strange !

But it's fine if you don't want it, nothing blocking for us (will use 'ext__' then.)

ThiefMaster commented:
yes, please keep the check :) With you guys I'm not worried that you properly test for breakage, but in the end this is a public API and I don't want others to do stupid things which then break and cause more work for us explaining them how to un-break them ;)

vtran99 commented:
Change updated with the check for existing field.
